utils/land-patterns.stanza

```diff
@@ -151,6 +151,34 @@ public defn rect-pth-pad (drill-r:Double, pad-w:Double, pad-h:Double) :
 public defn rect-pth-pad (drill-w:Double, drill-h:Double, pad-w:Double, pad-h:Double) :
   rect-pth-pad(C, drill-w, drill-h, pad-w, pad-h)
 
+pcb-pad roundrect-pth-pad (anchor:Anchor, drill-w:Double, drill-h:Double, pad-w:Double, pad-h:Double, pad-r:Double) :
+  name = to-string("%_ RoundedRectangle PTH Pad (drill = (%_, %_), pad = (%_, %_, %_))" % [
+           anchor, drill-w, drill-h, pad-w, pad-h, pad-r])
+  type = TH
+  if drill-w != drill-h :
+    fatal("roundrect-pth-pad(): RoundedRectangle holes are not supported")
+  
+  shape = RoundedRectangle(pad-w, pad-h, pad-r)
```

Comment:
@CuppoJava we do not check validity of pad-w, pad-h and pad-r here, right? We let the RoundedRectangle 'constructor' handle it, right?

Reply:
If `drill-w != drill-h` then the cutout layer can be a capsule. `rect-pth-pad` and `oval-pth-pad` demonstrate that.

Reply:
Yes. It's fine to let the constructor of RoundedRectangle check any preconditions.

Reply:
Do we want it to be a capsule? I mean, the user asks for a rounded rectangle and we provide only a capsule.

Reply:
Is the question: how do we import a Kicad project that contains rounded rectangle drill holes, when we do not support it?

It should either be an error. (This KiCad project cannot be imported for the following reasons.) Or the Kicad importer actually performs a sensible replacement (but this must be made absolutely precise and documented.)

Reply:
yes exactly. I would vote for an error (blame the caller) when asking for a drill with width different from height. I acknowledge it is currently done as a sensible replacement.

Reply:
Hmmm, okay I just read through the context some more:

This function happens to be used by the importer, but it is also a standalone OCDB utility that declares a certain pad type. I believe it is common to have rounded-rectangular pads with a capsule-shaped drill hole in the middle. (Please verify with Steve/Duncan, but I believe rounded-rectangular holes are very very rare.) 

So my vote is this pad declaration actually creates a rounded-rectangular pad, but with a capsule-shaped drill hole.

Reply:
👍

Reply:
Rounded rectangles are primarily used for surface mount pads. If a through hole pad uses a rounded rectangle, it will likely take a round hole, or a slotted hole. Of course, there's always corner cases for weird connectors

Reply:
'Slotted hole' is probably better user level syntax than capsule hole. Should rounded-rectangle just subsume capsules? Capsule seems to just be a special case of a rounded rectangle where r = width/2